src/core/buffer/events_generators.ts

```diff
@@ -0,0 +1,198 @@
+  IPeriodBufferReadyEvent,
+  IRepresentationChangeEvent,
+  IResumeStreamEvent,
+} from "./types";
+
+const EVENTS = {
+  activeBuffer(bufferType: IBufferType) : IBufferStateActive {
+    return {
+      type: "active-buffer",
+      value: { bufferType },
+    };
+  },
+
+  activePeriodChanged(period : Period) : IActivePeriodChangedEvent {
+    return {
+      type : "activePeriodChanged",
+      value : {
+        period,
+      },
+    };
+  },
+
+  adaptationChange(
+    bufferType : IBufferType,
+    adaptation : Adaptation|null,
+    period : Period
+  ) : IAdaptationChangeEvent {
+    return {
+      type: "adaptationChange",
+      value : {
+        type: bufferType,
+        adaptation,
+        period,
+      },
+    };
+  },
+
+  addedSegment<T>(
+    bufferType : IBufferType,
+    segment : ISegment,
+    segmentData : T
+  ) : IBufferEventAddedSegment<T> {
+    return {
+      type : "added-segment",
+      value : { bufferType, segment, segmentData },
+    };
+  },
+
+  bitrateEstimationChange(
+    type : IBufferType,
+    bitrate : number|undefined
+  ) : IBitrateEstimationChangeEvent {
+    return {
+      type: "bitrateEstimationChange",
+      value: {
+        type,
+        bitrate,
+      },
+    };
+  },
+
+  bufferComplete(bufferType: IBufferType) : ICompletedBufferEvent {
+    return {
+      type: "complete-buffer",
+      value: {
+        type: bufferType,
+      },
+    };
+  },
+
+  discontinuityEncountered(
+    bufferType : IBufferType,
+    nextTime : number
+  ) : IBufferNeedsDiscontinuitySeek {
+    return {
+      type : "discontinuity-encountered",
+      value : { bufferType, nextTime },
+    };
+  },
+
+  endOfStream() : IEndOfStreamEvent {
+    return {
+      type: "end-of-stream",
+      value: undefined,
+    };
+  },
+
+  fullBuffer(bufferType : IBufferType) : IBufferStateFull {
+    return {
+      type: "full-buffer",
+      value: { bufferType },
+    };
+  },
+
+  needsManifestRefresh(bufferType : IBufferType) : IBufferNeedsManifestRefresh {
+    return {
+      type : "needs-manifest-refresh",
+      value : { bufferType },
+    };
+  },
+
+  periodBufferReady(
+    type : IBufferType,
+    period : Period,
+    adaptation$ : Subject<Adaptation|null>
+  ) : IPeriodBufferReadyEvent {
+    return {
+      type: "periodBufferReady",
+      value: {
+        type,
+        period,
+        adaptation$,
+      },
+    };
+  },
+
+  periodBufferCleared(
+    type : IBufferType,
+    period : Period
+  ) : IPeriodBufferClearedEvent {
+    return {
+      type: "periodBufferCleared",
+      value: {
+        type,
+        period,
+      },
+    };
+  },
+
+  representationChange(
+    type : IBufferType,
+    period : Period,
+    representation : Representation
+  ) : IRepresentationChangeEvent {
+    return {
+      type: "representationChange",
+      value: {
+        type,
+        period,
+        representation,
+      },
+    };
+  },
+
+  resumeStream() : IResumeStreamEvent {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 119. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=283&issues=AWUoCSgDOTG07Bqb98zG&open=AWUoCSgDOTG07Bqb98zG)

[ISSUE_KEY]: AWUoCSgDOTG07Bqb98zG